src/sb_utils.R

```diff
@@ -29,4 +47,58 @@ sb_render_post_xml <- function(sb_id, ..., xml_file = NULL){
   
   sb_replace_files(sb_id = sb_id, xml_file)
   
-}
\ No newline at end of file
+}
+
+# Helper function to create a task_table for the files that need to be pushed to SB
+do_item_replace_tasks <- function(sb_id, files) {
+  
+  # Define task table columns
+  sb_push <- scipiper::create_task_step(
+    step_name = 'push_file_to_sb',
+    target_name = function(task_name, step_name, ...){
+      sprintf("%s_pushed_to_sb", task_name)
+    },
+    command = function(task_name, ...){
+      sprintf("upload_and_record(I('%s'), '%s')", sb_id, task_name)
+    } 
+  )
+  
+  # Create the task plan
+  task_plan <- create_task_plan(
+    task_names = files, # Define task table rows
+    task_steps = list(sb_push),
+    final_steps = c('push_file_to_sb'),
+    add_complete = FALSE)
+  
+  # Create the task remakefile
+  task_yml <- "file_upload_tasks.yml"
+  final_target <- "upload_timestamps"
+  create_task_makefile(
+    task_plan = task_plan,
+    makefile = task_yml,
+    packages = c('sbtools', 'scipiper', 'purrr'),
+    final_targets = final_target,
+    finalize_funs = "combine_upload_times",
+    as_promises = FALSE)
+  
+  # Build the tasks
+  loop_tasks(task_plan = task_plan, task_makefile = task_yml, num_tries = 3)
+  upload_timestamps <- remake::fetch(final_target, remake_file=task_yml)
+  
+  # Remove the temporary task makefile for uploading the files to ScienceBase
+  file.remove(task_yml)
+  
+  return(upload_timestamps)
+}
+
+upload_and_record <- function(sb_id, file) {
+  # First, upload the file
+  item_replace_files(sb_id, file)
+  
+  # Then record when it happened and return that as an obj
+  return(tibble(file = file, sb_id = sb_id, time_uploaded_to_sb = Sys.time()))
```

Comment:
can you format Sys.time() as a string and also convert to UTC so the answer is the same in different timezones?